### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by carlopi**

### CODE REVIEW

**Comment by carlopi** [comment]
This should fix the warnings detected by CRAN + other minor fixes.

On https://github.com/duckdb/duckdb/pull/6036/commits/4e595b1c0ddefa1cdc9c609b5745b6cd50e317c9, adding the initialization should do no harm, but valgrid signals that sometimes on current master it gets to the D_ASSERT(state.version == version) with uninitialized data.
```
idx_t ColumnData::ScanVector(ColumnScanState &state, Vector &result, idx_t remaining) {
        state.previous_states.clear();
        if (state.version != version) {
                InitializeScanWithOffset(state, state.row_index);
                state.current->InitializeScan(state);
                state.initialized = true;
        } else if (!state.initialized) {
                D_ASSERT(state.current);
                state.current->InitializeScan(state);
                state.internal_index = state.current->start;
                state.initialized = true;
        }
        D_ASSERT(data.HasSegment(state.current));
        D_ASSERT(state.version == version);
        // more stuff
```
@Mytherin: Should this not be inverted, with the check on initialization being done first, and only comparing version between initialized values?

File: .github/workflows/Regression.yml [hunk 1]
```diff
@@ -66,31 +66,31 @@ jobs:
...
     - name: Regression Test Micro
-      continue-on-error: true
+      if: ${{ always() }}
       shell: bash
       run: |
         python scripts/regression_test_runner.py --old=duckdb/build/release/benchmark/benchmark_runner --new=build/release/benchmark/benchmark_runner --benchmarks=.github/regression/micro.csv --verbose --threads=2
 
     - name: Regression Test TPCH
-      continue-on-error: true
+      if: ${{ always() }}
```
Comment (Mause) [inline .github/workflows/Regression.yml:75]: Nitpick, but these can be simplified to `if: always()` 😉

Comment (carlopi) [reply .github/workflows/Regression.yml]: Cleaned up, thanks!

File: third_party/fmt/include/fmt/core.h [hunk 2]
```diff
@@ -356,7 +356,11 @@ using string_view = basic_string_view<char>;
 using wstring_view = basic_string_view<wchar_t>;
 
 // A UTF-8 code unit type.
+#if FMT_HAS_FEATURE(__cpp_char8_t)
```
Comment (Tmonster) [inline third_party/fmt/include/fmt/core.h:359]: haha, yes. Much more eloquent than what I had

**Review by Tmonster — APPROVED** [review]

**Review by Tmonster — APPROVED** [review]
lgtm!

File: third_party/libpg_query/src_backend_parser_gram.cpp [hunk 3]
```diff
@@ -17298,6 +17298,7 @@ YYLTYPE yylloc;
   yystate = 0;
   yyerrstatus = 0;
   yynerrs = 0;
+  (void)yynerrs;
```
[the remark this answers is not in the document]
Comment (carlopi) [reply third_party/libpg_query/src_backend_parser_gram.cpp:17301]: Thanks for the catch, should be addressed here: b20dc943d1a1cbd2c3e0d1f77b6ce3ceb00a6051 (commit on top).

**Pull request merged**